src/categorical_algebra/HomSearch.jl

```diff
@@ -0,0 +1,721 @@
+module HomSearch 
+export   ACSetHomomorphismAlgorithm, BacktrackingSearch, HomomorphismQuery,
+        homomorphism, homomorphisms, is_homomorphic,
+       isomorphism, isomorphisms, is_isomorphic,
+       @acset_transformation, @acset_transformations,
+       subobject_graph, partial_overlaps, maximum_common_subobject
+   
+using ...Theories, ..CSets, ..FinSets, ..FreeDiagrams, ..Subobjects
+using ...Graphs.BasicGraphs
+using ..CSets: map_components
+using ACSets.DenseACSets: attrtype_type, delete_subobj!
+
+using Random
+using CompTime
+using MLStyle: @match
+using DataStructures: BinaryHeap, DefaultDict
+
+
+"""
+Functionality related to search problems involving ACSets, e.g.:
+
+- enumerating Hom(X,Y) where X,Y are ACSets
+- enumerating subobjects of an ACSet, X
+- enumerating partial overlaps between ACSets
+"""
```

Comment:
This docstring can become the module-level docstring, i.e., be moved to the very top of the file.